### ISSUE TITLE
A training problem about  " Iteration 40, loss = 1.#QNAN" 

### ISSUE THREAD

**Issue by GaryWon** [alexgkendall/caffe-posenet#12]
Hi, I have some issue concerning the python wrapper in caffe-windows. For some reason, I should transplant the code and run in windows. After some toss, the script "create_posenet_lmdb_dataset.py" and "test_posenet.py" can run pass. Since "KingsCollege_test_lmdb" and "KingsCollege_train_lmdb" are created and the output of  "test_posenet.py" are as follow:
`...`
`Iteration:   344   Error XYZ (m):   0.759703   Error Q (degrees):   2.12587290039`
`Iteration:   345   Error XYZ (m):   0.957241   Error Q (degrees):   4.29497876379`
`Iteration:   345   Error XYZ (m):   0.957241   Error Q (degrees):   4.29497876379`
`Median error  1.91579926014 m  and  5.39524091714 degrees.`
`Success!`
( the code is run on a 8GB GPU: NVIDIA Quadro K5200. )
Seems it work. However, I try to train PoseNet with  KingsCollege dataset, the loss value become invalid after "Iteration 20":
I0920 17:24:59.279114  7664 net.cpp:283] Network initialization done.
I0920 17:24:59.280113  7664 solver.cpp:60] Solver scaffolding done.
I0920 17:25:02.902320  7664 solver.cpp:288] Solving GoogLeNet
I0920 17:25:02.902320  7664 solver.cpp:289] Learning Rate Policy: step
I0920 17:25:05.076445  7664 solver.cpp:237] Iteration 0, loss = 16.29
I0920 17:25:05.076445  7664 solver.cpp:253]     Train net output #0: loss1/loss1_wpqr = 0.00497414 (\* 150 = 0.746121 loss)
I0920 17:25:05.076445  7664 solver.cpp:253]     Train net output #1: loss1/loss1_xyz = 17.2635 (\* 0.3 = 5.17906 loss)
I0920 17:25:05.076445  7664 solver.cpp:253]     Train net output #2: loss2/loss2_wpqr = 0.00427971 (\* 150 = 0.641956 loss)
I0920 17:25:05.076445  7664 solver.cpp:253]     Train net output #3: loss2/loss2_xyz = 14.6876 (\* 0.3 = 4.40629 loss)
I0920 17:25:05.076445  7664 solver.cpp:253]     Train net output #4: loss3/loss3_wpqr = 0.00250141 (\* 500 = 1.25071 loss)
I0920 17:25:05.076445  7664 solver.cpp:253]     Train net output #5: loss3/loss3_xyz = 4.06588 (\* 1 = 4.06588 loss)
I0920 17:25:05.076445  7664 sgd_solver.cpp:106] Iteration 0, lr = 0.001
I0920 17:25:33.759085  7664 solver.cpp:237] Iteration 20, loss = 1.#QNAN
I0920 17:25:33.759085  7664 solver.cpp:253]     Train net output #0: loss1/loss1_wpqr = 1.#QNAN (\* 150 = 1.#QNAN loss)
I0920 17:25:33.759085  7664 solver.cpp:253]     Train net output #1: loss1/loss1_xyz = 1.#QNAN (\* 0.3 = 1.#QNAN loss)
I0920 17:25:33.759085  7664 solver.cpp:253]     Train net output #2: loss2/loss2_wpqr = 1.#QNAN (\* 150 = 1.#QNAN loss)
I0920 17:25:33.759085  7664 solver.cpp:253]     Train net output #3: loss2/loss2_xyz = 1.#QNAN (\* 0.3 = 1.#QNAN loss)
I0920 17:25:33.759085  7664 solver.cpp:253]     Train net output #4: loss3/loss3_wpqr = 1.#QNAN (\* 500 = 1.#QNAN loss)
I0920 17:25:33.759085  7664 solver.cpp:253]     Train net output #5: loss3/loss3_xyz = 1.#QNAN (\* 1 = 1.#QNAN loss)

Here is my code:
`import caffe`
`caffe.set_device(0)`
`caffe.set_mode_gpu()`
`solver = caffe.SGDSolver(r"../models/solver_posenet.prototxt")`
`solver.net.copy_from(r"../models/weights_KingsCollege.caffemodel")`
`solver.solve()`

Do I miss some important things when I move the code to windows or  I should adjust some parameters before training? My code is based on the caffe version [https://github.com/microsoft/caffe](url)

**Comment by Duke02** [comment]
I would think it's mostly because your code is not based on the code that Posenet uses, which is different from either version of Caffe (Microsoft's or the original). It's discussed [here](https://groups.google.com/forum/#!topic/caffe-users/N5ZkdoMXiJQ) in the original Caffe google groups. Hope this helps! (That is if this is even still relevant to you after 2 years. Oh well. Might help someone else.)

**Comment by DRAhmadFaraz** [comment]
@Duke02 Dear

According to your instructions I have successfully installed Caffe from your repository.
When I Run this command "Python create_posenet_lmdb_dataset.py" It shiows like this.

![screenshot from 2019-01-28 10-09-06](https://user-images.githubusercontent.com/38114191/51829107-d0a6f000-22e4-11e9-9d2b-f3171e491d65.png)

After that When i use this command "python test_posenet.py --model train_kingscollege.prototxt --weights weights_kingscollege.caffemodel --iter 100" I got error of some mean file missings as shown.

![screenshot from 2019-01-28 10-11-08](https://user-images.githubusercontent.com/38114191/51829263-195ea900-22e5-11e9-8f0e-92dd149f00e0.png)

I0128 10:10:24.872309 20176 layer_factory.hpp:77] Creating layer data
I0128 10:10:24.872444 20176 net.cpp:106] Creating Layer data
I0128 10:10:24.872455 20176 net.cpp:411] data -> data
I0128 10:10:24.872473 20176 net.cpp:411] data -> label
I0128 10:10:24.872486 20176 data_transformer.cpp:25] Loading mean file from: ../imagemean.binaryproto
F0128 10:10:24.872498 20184 db_lmdb.hpp:14] Check failed: mdb_status == 0 (2 vs. 0) No such file or directory
*** Check failure stack trace: ***
F0128 10:10:24.872499 20176 io.cpp:54] Check failed: fd != -1 (-1 vs. -1) File not found: ../imagemean.binaryproto
*** Check failure stack trace: ***
Aborted (core dumped)

Kindly help me. I ll be thankful to you.
